### PR TITLE
Federated sign in warning

### PR DESCRIPTION
**Pull request by powerful23**
*Issue #, if available:*
Check the current user first before doing `Auth.federatedSignIn`
*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/auth/src/Auth.ts [hunk 1]
```diff
@@ -1305,24 +1305,24 @@ export default class AuthClass {
...
+    public async federatedSignIn(
         provider: 'google'|'facebook'|'amazon'|'developer'|string, 
         response: FederatedResponse, 
         user: FederatedUser
     ): Promise<ICredentials>{
+        // To check if the user is already logged in
+        try {
+            const loggedInUser = await this.currentAuthenticatedUser();
+            logger.warn(`There is already a signed in user: ${loggedInUser} in your app. 
+                You should not call Auth.federatedSignIn method again as it may cause some unexpected behavior.`);
```
Comment (undefobj) [inline packages/auth/src/Auth.ts]: Remove "some" from "may cause some unexpected behavior"

File: packages/auth/src/Auth.ts [hunk 2]
```diff
@@ -1322,24 +1322,24 @@ export default class AuthClass {
...
-                logger.debug('federated sign in credentials', cred);
-                res(cred);
-                return;
-            }).catch(e => {
-                rej(e);
-                return;
-            });
-        });    
+        const credentials = await Credentials.set({ provider, token, identity_id, user, expires_at }, 'federation');
+        const currentUser = await this.currentAuthenticatedUser();
```
Comment (manueliglesias) [inline packages/auth/src/Auth.ts]: you already called `const loggedInUser = await this.currentAuthenticatedUser();` above inside the try catch, can you refactor this, something like:

```typescript
let currentUser;

try {
  currentUser = await this.currentAuthenticatedUser();
  // ... logger.warn
} catch (e) {}

// ...

dispatchAuthEvent('signIn', currentUser);
```

File: packages/auth/src/Auth.ts (hunk 1 above)
Comment (powerful23) [reply packages/auth/src/Auth.ts]: Done.

File: packages/auth/src/Auth.ts (hunk 2 above)
Comment (powerful23) [reply packages/auth/src/Auth.ts]: Added some comments to explain the code.

**Pull request merged**